### DISCUSSION TITLE
DMA1_Stream6

### DISCUSSION THREAD

**Discussion by SriramKodey** [bitcraze/discussions#1472] [Q&A] [answered]
Hi all,
I noticed that in motors.c DMA1_Stream6_IRQHandler is used to start TIM2_CH2. I want to use DAC with DMA on the crazyflie and the DAC_Channel2 is configured to DMA1_Stream6_CH7. I want to use a different implementation of DMA1_Stream6_IRQHandler. Do you have any suggestions on how I can do this?

Thanks!!

**Comment by evoggy** [comment]
Hi,

From what I can see this is only used if DSHOT is configured (i.e `CONFIG_MOTORS_ESC_PROTOCOL_DSHOT` is defined) which would only be the case on the Crazyflie Bolt or Brushless (i.e using brushless motors). Is this the case? If not the conflict will be on [UART2 if DMA is used](https://github.com/bitcraze/crazyflie-firmware/blob/9ee56072ec1a769ab6dac3e1a1a781ef105853ba/src/drivers/src/uart2.c#L281), which you will need to handle.

**Reply by SriramKodey** [reply]
Yes, I do have the conflict with uart2 DMA, I added another #ifndef conditional statement in the uart2 file and added a define in the stm32fxxx.h to force my definition of the DMA1_Stream6_IRQHandler to be invoked. I still don't see the desired DAC Output on the PA5 pin, not sure if my IRQHandler is the issue. I also wanted to know how I can add configs like `CONFIG_MOTORS_ESC_PROTOCOL_DSHOT`.

**Reply by evoggy** [reply, marked as answer]
There's some [examples here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/create_platform/) on how to modify the configuration to add options. Note that you do not have to add a new platform, it's just to show how to edit the files.
